src/app/Plans/components/PlanStatus.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { IMigration, IPlan } from '@app/queries/types';
+import { StatusIcon, StatusType } from '@konveyor/lib-ui';
+import { Progress, ProgressVariant } from '@patternfly/react-core';
+
+interface IPlanStatusProps {
+  plan: IPlan;
+  migration: IMigration;
+}
+
+const PlanStatus: React.FunctionComponent<IPlanStatusProps> = ({
+  plan,
+  migration,
+}: IPlanStatusProps) => {
+  if (plan.status.conditions.every((condition) => condition.type === 'Ready')) {
+    return <StatusIcon status={StatusType.Ok} label="Ready" />;
+  } else {
+    // TODO: This is only a placeholder more status check needs to be done once clarity with statuses.
+    const totalVMs = plan.spec.vmList.length;
+    const percentVMsDone = totalVMs > 0 ? (migration.status.nbVMsDone * 100) / totalVMs : 0;
+    const label = 'Running';
+
+    return (
+      <>
+        <StatusIcon status={StatusType.Warning} label={label} />
+        <Progress
```

Comment:
I think to get the status icon and the "X of Y migrated" label on the same line like the mockups, you'll need to wrap the StatusIcon in a `<Level>` layout with the StatusIcon and the label as `<LevelItem>`s. I don't think we can use the `label` prop of Progress like this, not sure.

Alternatively if there's some way to get the `label` for `Progress` to appear above the bar, we could stick the `Level` with both things in there.

Reply:
@mturley, yeah this one definitely needs more work, I was going to leave it to cover it with another PR once we have more details about the plan migration statuses from API. That said the I'll fix it with `Level` in the meantime.